File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/api/versioning/versioning.py
"""
Middleware de versioning API threed (Python) - Ultra avancé
Gère la détection, la validation, la logique métier multi-version
Utilisation : app.before_request(versioning_middleware)
"""
from flask import g, request

SUPPORTED_VERSIONS = ['1.0', '2.0']


def versioning_middleware():
    version = request.headers.get('X-Api-Version', '1.0')
    if version not in SUPPORTED_VERSIONS:
        from flask import abort, jsonify
        abort(400, description=f'API version not supported. Supported: {SUPPORTED_VERSIONS}')
    g.api_version = version
    if not hasattr(g, 'business_context'):
        g.business_context = {}
    g.business_context['api_version'] = version

File: backup_before_industrialisation_20250626/backend/metier_generator/templates/threed/api/versioning/versioning.js
// Middleware de versioning API threed (Node.js) - Ultra avancé
// Gère la détection, la validation, la logique métier multi-version
// Utilisation : app.use(require('./versioning'));

const SUPPORTED_VERSIONS = ['1.0', '2.0'];

function versioningMiddleware(req, res, next) {
  const version = req.headers['x-api-version'] || '1.0';
  if (!SUPPORTED_VERSIONS.includes(version)) {
    return res.status(400).json({ error: 'API version not supported', supported: SUPPORTED_VERSIONS });
  }
  req.apiVersion = version;
  req.businessContext = req.businessContext || {};
  req.businessContext.apiVersion = version;
  next();
}

module.exports = versioningMiddleware;

// Exemple d'intégration :
// app.use(require('./versioning'));
// Dans un contrôleur : req.apiVersion
